ReleaseProcessAutomation.UnitTests/Steps/JiraSubStepTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Moq;
+using NUnit.Framework;
+using ReleaseProcessAutomation.Configuration.Data;
+using ReleaseProcessAutomation.Jira;
+using ReleaseProcessAutomation.Jira.ServiceFacadeImplementations;
+using ReleaseProcessAutomation.ReadInput;
+using ReleaseProcessAutomation.SemanticVersioning;
+using ReleaseProcessAutomation.Steps.SubSteps;
+using Spectre.Console.Testing;
+
+namespace ReleaseProcessAutomation.UnitTests.Steps;
+
+[TestFixture]
+public class JiraSubStepTests
+{
+  private Mock<IJiraVersionReleaser> _jiraReleaserMock;
+  private Mock<IJiraVersionCreator> _jiraVersionCreatorMock;
+  private Mock<IInputReader> _inputReaderMock;
+  private Mock<IJiraIssueService> _jiraIssueServiceMock;
+  private TestConsole _console;
+  private Configuration.Data.Config _config;
+
+  [SetUp]
+  public void Setup ()
+  {
+    _jiraVersionCreatorMock = new Mock<IJiraVersionCreator>();
+    _jiraReleaserMock = new Mock<IJiraVersionReleaser>();
+    _jiraVersionCreatorMock.Setup(_ => _.CreateNewVersionWithVersionNumber(It.IsAny<string>()));
+    _jiraReleaserMock.Setup(_ => _.ReleaseVersion(It.IsAny<string>(), false));
+    _jiraReleaserMock.Setup(
+        _ => _.ReleaseVersionAndSquashUnreleased(It.IsAny<string>(), It.IsAny<string>()));
+    _jiraIssueServiceMock = new Mock<IJiraIssueService>();
+
+    _inputReaderMock = new Mock<IInputReader>();
+
+    _console = new TestConsole();
+    _config = new Configuration.Data.Config();
+    _config.Jira = new JiraConfig();
+    _config.Jira.StringUseNTLM = "true";
+    _config.Jira.JiraURL = "https://www.JiraURL.com/url";
+    _config.Jira.JiraProjectKey = "JiraProjectKey";
+  }
+
+  [Test]
+  public void Execute_CreatesTwoVersionsCorrectly_ThrowsNothing ()
+  {
+    var currentVersion = new SemanticVersion();
+    var nextVersion = new SemanticVersion { Patch = 1 };
+
+    _jiraVersionCreatorMock.Setup(_ => _.CreateNewVersionWithVersionNumber(currentVersion.ToString()))
+        .Returns("current").Verifiable();
+    _jiraVersionCreatorMock.Setup(_ => _.CreateNewVersionWithVersionNumber(nextVersion.ToString()))
+        .Returns("next").Verifiable();
+
+    var jiraSubStep = new JiraSubStep(
+        _console,
+        _inputReaderMock.Object,
+        _jiraIssueServiceMock.Object,
+        _jiraVersionCreatorMock.Object,
+        _jiraReleaserMock.Object);
+
+    Assert.That(() => jiraSubStep.Execute(currentVersion, nextVersion, true), Throws.Nothing);
+
+    _jiraVersionCreatorMock.Verify();
```

Comment:
Looking at this test I wondered why it would succeed but loose mock are apparently very loose in their behavior ^^

It made me realize two things:
 - I think we should make the input explicit (strict mock) and also check the output
 - If there are no issues to move we probably don't need to ask the user if they want to move issues